reactioncommerce/reaction-admin
# GraphQL-powered Accounts page

Resolves #230, resolves #283
Impact: **critical**
Type: **feature|bugfix|performance|style|refactor**

**Test with:**

- [api-plugin-email-templates#7](https://github.com/reactioncommerce/api-plugin-email-templates/pull/7)
- [api-plugin-accounts#8](https://github.com/reactioncommerce/api-plugin-accounts/pull/8)
- [api-plugin-accounts#10](https://github.com/reactioncommerce/api-plugin-accounts/pull/10)
- [api-plugin-accounts#14](https://github.com/reactioncommerce/api-plugin-accounts/pull/14)
- [api-plugin-accounts#15](https://github.com/reactioncommerce/api-plugin-accounts/pull/15)
- [api-plugin-accounts#16](https://github.com/reactioncommerce/api-plugin-accounts/pull/16)

All `api-plugin-accounts` PRs are available together on the `outgrow-2.0.0-next` branch.

## Issue
The Accounts page was half-working and was running on pure Meteor. It needed to be ported over to GraphQL, with all the functionalities working and possibly some requested additions.

## Solution
This PR proposes a new Accounts page that relies only on GraphQL and uses the new Catalyst design language. It revives some features that weren't working anymore for a long time, for example the ability to create new groups and to edit the permissions of existing ones. It also enables users to bulk-edit permissions for users.

With no specific directions for the design, I kind of freestyled my way into it. This is what it looks like:

<img width="1680" alt="Screen Shot 2020-05-02 at 23 56 50" src="https://user-images.githubusercontent.com/45489203/80893331-c044a880-8cd1-11ea-9bb9-90e498044fda.png">
<img width="1680" alt="Screen Shot 2020-05-02 at 23 57 00" src="https://user-images.githubusercontent.com/45489203/80893335-c2a70280-8cd1-11ea-8497-b2c34e701459.png">
<img width="1680" alt="Screen Shot 2020-05-02 at 23 57 14" src="https://user-images.githubusercontent.com/45489203/80893338-c470c600-8cd1-11ea-9bec-a3ab2b313b9f.png">
<img width="1680" alt="Screen Shot 2020-05-03 at 00 04 16" src="https://user-images.githubusercontent.com/45489203/80893341-c6d32000-8cd1-11ea-8761-7c05b607c7d7.png">
<img width="1680" alt="Screen Shot 2020-05-03 at 01 30 16" src="https://user-images.githubusercontent.com/45489203/80894713-baed5b00-8cdd-11ea-8abb-624081fca34b.png">
<img width="1675" alt="Screen Shot 2020-05-08 at 15 39 36" src="https://user-images.githubusercontent.com/45489203/81411104-2c943180-9142-11ea-9311-31208093a35c.png">

## Breaking changes
None.

## Testing
1. Go to the Accounts page with an owner account.
2. Try to invite a new staff member.
3. Sign up as this staff member on another, incognito tab.
4. Back to your owner account, create a new group.
5. Assign this new group to your new staff member.
6. Confirm in the `Accounts` MongoDB collection that the changes take effect.
7. Try to edit the name of this new group.
8. Invite another user and try to bulk-edit group attributions.
